crud/common/utils.lua

```diff
@@ -97,6 +97,20 @@ end
 
 function utils.get_space(space_name, replicasets)
```

Comment:
While expecting `utils.get_space` use cases, I've met some unsettling code: there are examples of
```lua
    local space = utils.get_space(space_name, vshard_router:routeall())
    if space == nil then
        return nil, LenError:new("Space %q doesn't exist", space_name)
    end
```
and examples of
```lua
    local space, err = utils.get_space(space_name, vshard_router:routeall())
    if err ~= nil then
        return nil, ReplaceError:new("Failed to get space %q: %s", space_name, err), const.NEED_SCHEMA_RELOAD
    end

    if space == nil then
        return nil, ReplaceError:new("Space %q doesn't exist", space_name), const.NEED_SCHEMA_RELOAD
    end
```
but `utils.get_space` never returns errors (at least in current versions). I think we may rework `utils.get_space` to work as follows:
- it's either returns `space` if everything is alright
- or it returns `nil, err` if something went wrong, like
    - missing master,
    - broken master connection,
    - missing space with error message similar to the current one.

This approach should make existing code more consistent, solve #331 and do not break existing behavior (maybe it will change error messages a bit, but regex error checks in tests should succeed nonetheless).

Reply:
I've reworked `utils.get_space`, now it returns two values:
* space
* err (instance of GetSpaceError:new(error_msg))

In my current patch the error `"Space %q doesn't exist"`  is not processed through `err`. 
To handle this error, `space == nil` is checked, and after that, an instance of the error is created, depending on the location in the code (for example, `SelectError:new('Space %q doesn't exist ...')` in `select.lua`, or `CountError:new('Space %q doesn't exist ...')` in `count.lua`)

At the same time, now all `utils.get_space`  calls process both `space` and `err` returns.
In specific places `const.NEED_SCHEMA_RELOAD` is used